### DISCUSSION TITLE
[CI] Unable to build Windows-Project with `dotnet publish`

### DISCUSSION THREAD

**Discussion by TopperDEL** [unoplatform/uno#12471] [Q&A] [answered]
I have an up-to-date Uno-project and I am trying to get a CI-pipeline set-up. I use the following command to build the Windows-Head:

`dotnet publish myApp.Windows/myApp.Windows.csproj -c Release -f net7.0-windows10.0.19041.0 --no-restore`

The app runs when built with Visual Studio and the prerequisits like Net 7 are installed. But I get an error that does not really lead me to anything:

`C:\Users\runneradmin\.nuget\packages\microsoft.windowsappsdk\1.1.3\buildTransitive\MrtCore.PriGen.targets(911,5): error MSB4062: The "Microsoft.Build.Packaging.Pri.Tasks.ExpandPriContent" task could not be loaded from the assembly C:\Users\runneradmin\AppData\Local\Microsoft\dotnet\sdk\7.0.302\\Microsoft\VisualStudio\v17.0\AppxPackage\\Microsoft.Build.Packaging.Pri.Tasks.dll. Could not load file or assembly 'C:\Users\runneradmin\AppData\Local\Microsoft\dotnet\sdk\7.0.302\Microsoft\VisualStudio\v17.0\AppxPackage\Microsoft.Build.Packaging.Pri.Tasks.dll'. The system cannot find the path specified. Confirm that the <UsingTask> declaration is correct, that the assembly and all its dependencies are available, and that the task contains a public class that implements Microsoft.Build.Framework.ITask.`

I found hints that `dotnet publish` is not (yet) available for MAUI-projects or projects using the Windows App SDK. But since general availability of Net 7 and MAUI this should be supported.

Any ideas what might cause this issue? Where could I start to further investigate?

**Comment by jeromelaban** [comment, marked as answer]
dotnet publish is not fully supported by WinAppSDK at this time: https://github.com/microsoft/WindowsAppSDK/issues/3548

**Reply by TopperDEL** [reply]
Weird. I was referencing [this](https://blog.taranissoftware.com/build-net-maui-apps-with-github-actions) blog post where it mentions that an MSBuild-Fallback is no more necessary.
I will try again with the solution mentioned in the issue linked above, but I think I tried this already and it did not work.

**Reply by TopperDEL** [reply]
I had to add the property mentioned in the issue to the uno-library-project to make it work. It works now - thanks!
